Sources/NIOCore/AsyncChannel/AsyncChannelOutboundWriterHandler.swift

```diff
@@ -96,15 +95,8 @@ internal final class NIOAsyncChannelOutboundWriterHandler<OutboundOut: Sendable>
     func _didTerminate(error: Error?) {
         self.eventLoop.preconditionInEventLoop()
 
-        switch self.closeRatchet.closeWrite() {
-        case .nothing:
-            break
-
-        case .closeOutput:
+        if self.isOutboundHalfClosureEnabled {
```

Comment:
Call out for reviewers: We are still closing the output side if users are calling `NIOAsyncWriter.finish` and outbound half closure is enabled. This is not triggered by deinit based cleanup